Include/pymacconfig.h

```diff
@@ -74,8 +77,14 @@
 #  define DOUBLE_IS_LITTLE_ENDIAN_IEEE754
 #endif
 
-#ifdef __i386__
+#if defined(__i386__) || defined(__x86_64__)
 #  define HAVE_GCC_ASM_FOR_X87
+#  define ALIGNOF_MAX_ALIGN_T 16
+#  define HAVE_GCC_ASM_FOR_X64 1
+#  define SIZEOF_LONG_DOUBLE 16
+#else
+#  define ALIGNOF_MAX_ALIGN_T 8
+#  define SIZEOF_LONG_DOUBLE 8
```

Comment:
Another issue is ``HAVE_GCC_ASM_FOR_X87``, which was silently disabled on macOS for both universal2 and intel-only builds (at least with Xcode 15.1, I haven't checked older Apple compiler).

If I read the code correctly ``ALIGNOF_MAX_ALIGN_T`` is only used in typeobject.c in the definition of ``_align_up`` and that's to get the correct alignment for type data that's specified with a negative item size which is a feature new in 3.12.   This likely only affects performance, although unless the subtype specific data contains vector types (AFAIK those have strict alignment requirements).

I'm not sure how to effectively search for external users of the macro. 

My gut feeling is that this should be save. Maybe ask Thomas about his opinion as RM?

Reply:
BTW. I get no hits on SIZEOF_LONG_DOUBLE at all in 3.13, other than exporting it through pyconfig and sysconfig.

Reply:
To be clear, one reason for backporting is that, without it, tests fail when universal2 builds are built on Apple Silicon Macs and then run on Intel Macs.

Reply:
I'll back port to 3.12 and propose to skip the back port to 3.11 because the impact there is likely lower:

- ALIGNOF_MAX_ALIGN_T isn't present in 3.11( added for a feature in 3.12)
- SIZEOF_LONG_DOUBLE is present, but isn't used by CPython itself
- HAVE_GCC_ASM_FOR_X64 is present, but isn't used by CPython itself
- HAVE_GCC_ASM_FOR_X87 is present and used, but X87_DOUBLE_ROUNDING is not set 

For the latter: I checked with the configure check for x87 double rounding that double rounding isn't detected when compiling for x86_64 using ``clang -arch x86_64``.

https://github.com/python/cpython/blob/1eac0aab859516378b64552b44b543a0dfd373c2/configure.ac#L5469-L5492